src/storage/txn/commands/prewrite.rs

```diff
@@ -2189,7 +2189,7 @@ mod tests {
         );
         assert!(matches!(
             err,
-            MvccError(box MvccErrorInner::AssertionFailed { .. })
+            MvccError(box MvccErrorInner::WriteConflict { .. })
```

Comment:
Is there a case when the assertion fails on the index key but there is no write conflict now?

Reply:
I can't think of such a case.
We've only encountered 2 false assertion failures, both of which should return write conflicts.